gems/aws-sdk-core/lib/aws-sdk-core/plugins/telemetry.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+module Aws
+  module Plugins
+    # @api private
+    class Telemetry < Seahorse::Client::Plugin
+      option(
+        :telemetry_provider,
+        default: Aws::Telemetry::NoOpTelemetryProvider,
+        doc_type: Aws::Telemetry::TelemetryProviderBase,
+        rbs_type: Aws::Telemetry::TelemetryProviderBase,
+        docstring: <<~DOCS) do |_cfg|
+Allows you to provide a telemetry provider, which is used
+to emit telemetry data. By default, uses `NoOpTelemetryProvider`
+which will not record or emit any telemetry data. The SDK currently supports
+OpenTelemetry (OTel) as a telemetry provider. To use the OTel provider, 
```

Comment:
I'd convert this to a list of .. one thing .. that can be added to later.

Reply:
I will do it now to set up for future :)